### DISCUSSION TITLE
MatEx integration with HotSniper

### DISCUSSION THREAD

**Discussion by anuiiita** [anujpathania/HotSniper#35] [General]
Hi community,
       I am trying to replace Hotspot with MatEx in HotSniper. In order to do so I am trying to run Matex with the temperature, power, floorplan and eigendata files which we have in Hotsniper, Matex run successfully but give no transient temperature in o/p. Below are the few details.

- For initial temperature I am using Temperature.init file from Hotsniper, and mentioning the same in matex.config.
- For power trace I am using PeriodicPower.log file from Hotsniper.
- eignedata and floorplan also I am using from Hotsniper.

Exact file names are = **PeriodicPower.log, Temperature.init, 8x8_eigendata.bin, 8x8_manycore.flp**
Matex command = `./MatEx -c matex.config -f 8x8_manycore.flp -p manycore.ptrace` Note: Here I just renamed PeriodicPower.log to manycore.ptrace
Change in matex.config:
 ```
# initial temperatures from file
-init_file			(null)
# initial temperature (kelvin) if not from file
-init_temp			333.15
```
TO
```
# initial temperatures from file
-init_file			Temperature.init
# initial temperature (kelvin) if not from file
-init_temp			333.15
```

Please help me here, Running matex with given floorplan, eigendata, powertrace and initial temperature would be really helpful for me.

**Comment by anujpathania** [comment]
Hi,

Out of curiosity, why do you want to replace HotSpot with MatEx?

**Reply by anuiiita** [reply]
Yes Sir, I want the transient temperature for my use case. I saw in few paper it is mentioned that MatEx is more accurate than Hotspot.

**Comment by yixianUvA** [comment]
Hi anuiiita,
    Thanks for your interest about HotSniper.
    If you target to get the tranisient temperature from MatEx, you should generate the eigin value in Matex.  You can check out the readme file in Matex.

USING MATEX
-------------

To use MatEx, the application needs a configuration file (similar to the one used by HotSpot), a power trace file, and a floorplan or an eigeninput file.
When the floorplan is given, the RC thermal network is computed through HotSpot. When the eigeninput file is given, the eigenvectors, eigenvalues, and all required matrix are taken from the eigeninput file.

Some examples of how MatEx is used follow:
1- Compute the peak temperatures for a given floorplan:
	./MatEx -c matex.config -f floorplan.flp -p powerTrace.ptrace

2- Compute the peak temperatures for a given floorplan and save the results into a file:
	./MatEx -c matex.config -f floorplan.flp -p powerTrace.ptrace -o maxTemp.data

3- Compute all transient temperatures and the peak temperatures for a given floorplan:
	./MatEx -c matex.config -f floorplan.flp -p powerTrace.ptrace -transient_file_block allTemp.data

4- Compute the peak temperatures for a given floorplan and generate an eigeninput file for next use:
	./MatEx -c matex.config -f floorplan.flp -p powerTrace.ptrace -eigen_out eigendata.bin

5- Compute the peak temperatures for a given eigeninput file (faster than with a floorplan file):
	./MatEx -c matex.config -i eigendata.bin -p powerTrace.ptrace

6- Compute all transient temperatures and the peak temperatures for a given eigeninput file (faster than with a floorplan file):
	./MatEx -c matex.config -i eigendata.bin -p powerTrace.ptrace -transient_file_block allTemp.data

**Reply by anuiiita** [reply]
Hello Sir,
Thanks for the reply.
I went through the README file. MatEx is running fine with floorplan and power trace files which are already provided (multicore.flp and multicore.ptrace).

But when I am using my floorplan and power trace files from Hotsniper (8*8_manycore.flp, PeriodicPower.log) MatEx is giving wrong temperature value (-273). I have mentioned below matex command and content PeriodicPower.ptrace file which I am using.

Note: I did required changes in PeriodicPower.log, like adding time col and giving -1 value for first row of power trace and so on. I assume eigendata will be generated by MatEx for given floorplan.

Please correct me if my understanding is not right.

`./MatEx -c matex.config -f 8x8_manycore.flp -p PeriodicPower.ptrace`

```
time	Core0-TP	Core1-TP	Core2-TP	Core3-TP	Core4-TP	Core5-TP	Core6-TP	Core7-TP	Core8-TP	Core9-TP	Core10-TP	Core11-TP	Core12-TP	Core13-TP	Core14-TP	Core15-TP	Core16-TP	Core17-TP	Core18-TP	Core19-TP	Core20-TP	Core21-TP	Core22-TP	Core23-TP	Core24-TP	Core25-TP	Core26-TP	Core27-TP	Core28-TP	Core29-TP	Core30-TP	Core31-TP	Core32-TP	Core33-TP	Core34-TP	Core35-TP	Core36-TP	Core37-TP	Core38-TP	Core39-TP	Core40-TP	Core41-TP	Core42-TP	Core43-TP	Core44-TP	Core45-TP	Core46-TP	Core47-TP	Core48-TP	Core49-TP	Core50-TP	Core51-TP	Core52-TP	Core53-TP	Core54-TP	Core55-TP	Core56-TP	Core57-TP	Core58-TP	Core59-TP	Core60-TP	Core61-TP	Core62-TP	Core63-TP
-1	0.917593216208	5.18168068616	5.18184290824	5.18141707528	5.19257998216	5.19250394056	5.19262053768	5.1921794964	5.18647130696	5.1867805428	5.18690220936	5.18645609864	5.19305650952	5.19289428744	5.19231637128	5.19289935688	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208
0	0.917593216208	5.18475783624	5.18449422536	5.18454491976	5.17140493128	5.17155701448	5.17132382024	5.17117173704	5.17929804936	5.1785933972	5.17864916104	5.17929297992	5.1705583348	5.17025923784	5.17139986184	5.17059382088	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208
0.5	0.917593216208	5.16400354888	5.1643482708	5.1640187572	5.17733110664	5.176920482	5.17699145416	5.17688499592	5.16902736392	5.1691388916	5.16910340552	5.16930618312	5.17716888456	5.17689006536	5.17731082888	5.17722971784	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208
1	0.917593216208	5.19616407624	5.19571796552	5.1957027572	5.17530333064	5.1762614548	5.1751968724	5.1759319412	5.19126192776	5.19094762248	5.1909121364	5.190988178	5.175526386	5.17559228872	5.17561256648	5.17632735752	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208
1.5	0.917593216208	5.16864208648	5.16900708616	5.16846465608	5.18928484616	5.18846866632	5.18903137416	5.18793637512	5.17257090248	5.17308291592	5.17343777672	5.17335159624	5.18881845768	5.18808845832	5.18877283272	5.18815943048	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208	0.917593216208
```

Please let me know if am doing any mistake here.

**Comment by SobhanNiknam** [comment]
Hi

I think the problem is probably due to a bug in MatEx. The chip shouldn’t have indeed square shape. So, please slightly modify numbers in your floorplan to make the chip rectangular. Try your experiments again with the new floorplan and let us know about the outcome!

Best regards,
Sobhan

**Reply by anuiiita** [reply]
Hi All,
Looks like we can't supply our periodic power log file directly to MatEx.
I read periodic power file into my script and write again to new file by applying ('\t') separator it started working. Periodic power file also has '\t' separator but I am not sure why MatEx was not working when I was giving our periodic power log file directly(just adding time col).

Thanks all for the help, I will close this discussion now.
I will ask for help here in future if required :)
